portal/webapp/src/main/java/no/dcat/portal/webapp/PortalRestController.java

```diff
@@ -19,109 +22,299 @@
         try {
-            decodedUri = new java.net.URI(id).toString();
-        } catch (URISyntaxException e) {
+            id = URLDecoder.decode(id, "UTF-8");
+
+        } catch (UnsupportedEncodingException e) {
+            // Should in principle newer be thrown
             logger.error("URI syntax error ", id, e);
+            return null;
         }
 
-        String queryString =
-                "PREFIX dcat: <http://www.w3.org/ns/dcat#> " +
-                "PREFIX dct: <http://purl.org/dc/terms/> " +
-                "PREFIX owl: <http://www.w3.org/TR/owl-time/> " +
-                "PREFIX adms: <http://www.w3.org/ns/adms#>" +
-                "PREFIX foaf: <http://xmlns.com/foaf/0.1/>" +
-                "PREFIX vcard: <http://www.w3.org/2006/vcard/ns#>" +
-                "PREFIX dcatno: <http://difi.no/dcatno#>" +
-                "PREFIX er: <http://data.brreg.no/meta/>" +
-                "DESCRIBE ?dataset ?publisher ?contact ?distribution ?nace ?fadr ?padr ?sektor " +
-                "where {?dataset a dcat:Dataset. " +
-                " ?dataset dct:publisher ?publisher." +
-                        " OPTIONAL {?publisher er:naeringskode1 ?nace}" +
-                        " OPTIONAL {?publisher er:forretningsadresse ?fadr}" +
-                        " OPTIONAL {?publisher er:postadresse ?padr}" +
-                        " OPTIONAL {?publisher er:institusjonellSektorkode ?sektor}" +
-                " OPTIONAL {?dataset dcat:contactPoint ?contact} " +
-                " OPTIONAL {?dataset dcat:distribution ?distribution }" +
-                " FILTER (?dataset = <" + decodedUri + "> ) " +
-                "}";
-
-        Query query = QueryFactory.create(queryString);
-
-        try (QueryExecution qexec = QueryExecutionFactory.create(query, model)){
-            Model submodel = qexec.execDescribe();
+        Query query = getQuery(String.format(queryString, id));
+
+        try (QueryExecution qe = getQueryExecution(query)) {
+
+            logger.trace(query.toString());
+
+            Model submodel = qe.execDescribe();
+
             if (submodel.isEmpty()) {
-                throw new NoSuchElementException("Empty result");
+                return null;
             }
             ModelFormatter modelFormatter = new ModelFormatter(submodel);
 
             return modelFormatter.format(format);
+        } catch (Exception e) {
+            logger.error("No resource found {}", e.getClass().getName(), e);
+            return null;
         }
     }
 
+    /**
+     * Creates a query based on a sparql text query
+     *
+     * @param query the sparql query
+     * @return the query object
+     */
+    Query getQuery(String query) {
+        return QueryFactory.create(query);
+    }
+
+    /**
+     * Do query fuseki service
+     *
+     * @param query sparql query
+     * @return the execution handler of the query
+     */
+    QueryExecution getQueryExecution(Query query) throws Exception {
```

Comment:
Bamboo klassifiserer det som en "major bug" at getQueryExecution kaster en generell Exception i stedet for en dedikert exception.